std/functional.d

```diff
@@ -707,30 +707,55 @@ template partial(alias fun, alias arg)
     {
         static assert(false, "Cannot apply partial to a non-callable '" ~ fun.stringof ~ "'.");
     }
-    else // Assume fun is callable and uniquely defined.
+    else
     {
-        static if (Parameters!fun.length == 0)
+        import std.meta : Filter;
+
+        static if (__traits(compiles, __traits(getOverloads,
+            __traits(parent, fun), __traits(identifier, fun))))
```

Comment:
Alas, there doesn't seem to be a better way to do this. The problem is that `__traits(getOverloads, __traits(parent, fun), ...)` fails for static nested functions (among others), but `traits(isNested, fun)` returns `false` because static functions don't store a context pointer.